### ISSUE TITLE
Update Aurora's PyTorch DDP core affinity recommendations?

### ISSUE THREAD

**Issue by felker** [argonne-lcf/user-guides#901]
### Description

Following up on #865, the DDP recommended "Settings for training beyond 16 nodes"
https://docs.alcf.anl.gov/aurora/data-science/frameworks/pytorch/#code-changes-to-train-on-multiple-gpus-using-ddp

Option 2 might be missing `-d 4` argument to `mpiexec`?

e.g. an ALCC project training on 31 nodes using
```bash
mpiexec -n 372 --ppn 12 --cpu-bind=verbose,list:4-7:8-11:12-15:16-19:20-23:24-27:56-59:60-63:64-67:68-71:72-75:76-79 python ./app.py ...
```
where the application is using a `DataLoader` with `num_workers=4`, they get a warning 
```
UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 1.
```

If you add `--depth 4`, the warning goes away:
```bash
mpiexec -n 372 --ppn 12 -d4 --cpu-bind=verbose,list:4-7:8-11:12-15:16-19:20-23:24-27:56-59:60-63:64-67:68-71:72-75:76-79 python ./app.py ...
```

And they experience a sizable boost in throughput:
```
New configuration
Epoch 0: 5%|▌ | 16/294 [03:34<1:02:04, 13.40s/it
Original configuration
Epoch 0: 5%|▌ | 16/294 [05:28<1:35:15, 20.56s/it,
```

### Priority Level

Medium (Normal priority)

### Label Confirmation

- [x] I have selected at least one appropriate label
